synthesizer/process/src/finalize.rs

```diff
@@ -241,8 +240,17 @@ fn finalize_constructor<N: Network, P: FinalizeStorage<N>>(
     // Get the constructor types.
     let constructor_types = stack.get_constructor_types()?.clone();
 
+    // Set the FinalizeRegisters function name for constructors.
+    // Before ConsensusVersion::V12, this was wrongly using the program id as function name.
+    let consensus_version = N::CONSENSUS_VERSION(state.block_height())?;
+    let identifier = if consensus_version >= ConsensusVersion::V12 {
```

Comment:
If the purpose of this change is to improve the log message, can you pass in a flag to `finalize_command_expect_await` and set the message accordingly?

This avoids the permanent "overhead" of tracking this to a `ConsensusVersion`